ui/src/wizard/AzureWizard.ts

```diff
@@ -72,16 +86,31 @@ export class AzureWizard<T extends IInternalActionContext> implements types.Azur
                 if (step.shouldPrompt(this._context)) {
                     step.propertiesBeforePrompt = Object.keys(this._context).filter(k => !isNullOrUndefined(this._context[k]));
 
-                    const disposable: vscode.Disposable = this._context.ui.onDidFinishPrompt((result) => {
+                    const loadingQuickPick = this._showLoadingPrompt ? createQuickPick(this._context, {
+                        loadingPlaceHolder: localize('loading', 'Loading...')
+                    }) : undefined;
+
+                    const disposables: vscode.Disposable[] = [];
+
+                    if (loadingQuickPick) {
+                        disposables.push(loadingQuickPick?.onDidHide(() => {
+                            this._cancellationTokenSource.cancel();
+                        }));
+                    }
+
+                    disposables.push(this._context.ui.onDidFinishPrompt((result) => {
                         // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
                         step!.prompted = true;
+                        loadingQuickPick?.show();
+                        this._cancellationTokenSource = new vscode.CancellationTokenSource();
```

Comment:
First of all, each `CancellationTokenSource` should be disposed of properly. Secondly, I think there's a small chance of timing issues

I would actually approach this in a different way, though - rather than create a new token source each time, I would want to prevent the existing token source from cancelling itself when we don't want it to. In other words, `loadingQuickPick?.onDidHide` above would somehow need to know when `AzExtUserInput` is about to prompt so that they can ignore that specific event. Something along the lines of `AzExtUserInput.onDidStartPrompt: Event<whatever>` or `AzExtUserInput.isPrompting: boolean` would probably help accomplish that.